_posts/2021-12-31-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,348 @@
+* `DENY` statement, which enables you to remove a user or groups access via SQL.
+* `IN catalog` clause to `CREATE ROLE`, `DROP ROLE`, `GRANT ROLE`, 
+`REVOKE ROLE`, and `SET ROLE` to specify the target catalog of the statement 
+instead of using the current session catalog.
+* [`listagg()`]({{site.url}}/docs/current/functions/aggregate.html#listagg)
+function returns concatenated values seperated by a specified separator.
+* [`SET TIME ZONE`]({{site.url}}/docs/current/sql/set-time-zone.html) keywords.
+* Arbitrary queries in [`SHOW STATS`]({{site.url}}/docs/current/sql/show-stats.html).
+* Subquery expressions that return multiple columns. Example: `SELECT x = (VALUES (1, 'a'))`.
+* Add support for `ALTER MATERIALIZED VIEW` ... `RENAME TO` ...
+
+### Query processing improvements
+
+* Added support for automatic query retries (this feature is very experimental
+with some limitations for now).
+* Updated the behavior of `ROW` to `JSON` cast to produce `JSON` objects instead
+of `JSON` arrays.  
+
+## Performance improvements
+
+Improved performance for the following operations:
+
+* Querying Parquet data for files containing column indexes.
+* Reading dictionary-encoded Parquet files.
+* Queries using [`rank()`]({{site.url}}/docs/current/functions/window.html#rank) window function.
+* Queries using [`sum()`]({{site.url}}/docs/current/functions/aggregate.html#sum)
+and [`avg()`]({{site.url}}/docs/current/functions/aggregate.html#avg) for 
+decimal types.
+* Queries using `GROUP BY` with single grouping column.
+* Aggregation on decimal values.
+* Evaluation of the WHERE and SELECT clause.
+* Computing the product of decimal values with precision larger than 19.
+* Queries that process row or array data.
+* Queries that contain a `DISTINCT` clause. 
+* Reduced memory usage when performing joins.
+
+## Security
+
+Added the following improvements and features relevant for authentication, 
+authorization and integration with other security systems:
+
+* Automatic configuration of TLS for 
+[secure internal communication]({{site.url}}/docs/current/security/internal-communication.html).
+This removes the need to provision per-worker TLS certificates.
+* OAuth2/OIDC opaque access tokens.
+* Configuring multiple password authentication plugins.
+* Hiding inaccessible columns from `SELECT *` statement.
+* Configuring HTTP proxy for OAuth2 authentication.
+* Handling of Server Name Indication (SNI) for multiple TLS certificates.
+
+## Data Sources
+
+### BigQuery connector
+
+* Added support for case insensitive name matching for BigQuery views.
+* Support reading `bignumeric` type whose precision is less than or equal to 
+38.
+* Add support for `CREATE SCHEMA` and `DROP SCHEMA` statements.
+* Improved support for BigQuery datetime and timestamp types.
+* Fixed failures for queries accessing `information_schema.columns` when 
+`case-insensitive-name-matching` is disabled.
+
+### Cassandra connector
```

Comment:
Support for reading user-defined types. https://github.com/trinodb/trino/issues/147